wallet-new/test/unit/UTxO/BlockGen.hs

```diff
@@ -83,33 +89,38 @@ runBlockGenWith settings boot m =
 initializeCtx :: Hash h Addr => Transaction h Addr -> BlockGenCtx h
 initializeCtx boot = BlockGenCtx {..}
   where
-    _bgcCurrentUtxo = trUtxo boot
-    _bgcFreshHash = 1
-    _bgcInputPartiesUpperLimit = 1
+    _blockGenCtxCurrentUtxo = trUtxo boot
+    _blockGenCtxFreshHashSrc = 1
+    _blockGenCtxInputPartiesUpperLimit = 1
 
 -- | Lift a 'Gen' action into the 'BlockGen' monad.
 liftGen :: Gen a -> BlockGen h a
 liftGen = BlockGen . lift
 
 -- | Provide a fresh hash value for a transaction.
-freshHash :: BlockGen h Int
+freshHash :: ( HasFreshHashSrc src Int
+             , MonadState src m
+             )
+          => m Int
 freshHash = do
-    i <- use bgcFreshHash
-    bgcFreshHash += 1
+    i <- use freshHashSrc
+    freshHashSrc += 1
     pure i
 
-bgcNonAvvmUtxo :: Getter (BlockGenCtx h) (Utxo h Addr)
-bgcNonAvvmUtxo =
-    bgcCurrentUtxo . to (utxoRestrictToAddr (not . isAvvmAddr))
+nonAvvmUtxo :: HasCurrentUtxo src (Utxo h Addr)
+            => Getter src (Utxo h Addr)
+nonAvvmUtxo =
+    currentUtxo . to (utxoRestrictToAddr (not . isAvvmAddr))
 
-selectSomeInputs :: Hash h Addr => BlockGen h (NonEmpty (Input h Addr, Output Addr))
-selectSomeInputs = do
-    utxoMap <- uses bgcNonAvvmUtxo utxoToMap
-    upperLimit <- use bgcInputPartiesUpperLimit
-    input1 <- liftGen $ elements (Map.toList utxoMap)
+selectSomeInputs' :: Hash h Addr
+                  => Int
```

Comment:
What are the guidelines for dealing with integer numbers where only positives are expected to succeed?

Reply:
Changing to `Nat` or `Word` is a good call IMO.